src/vivarium/framework/results/interface.py

```diff
@@ -9,6 +9,27 @@
     from vivarium.framework.results.manager import ResultsManager
 
 
+def _raise_missing_unstratified_observation_results_gatherer(*args, **kwargs) -> pd.DataFrame:
```

Comment:
This is kind of clunky, but I created a very specific raise function for each default handling type to provide a more useful error message (we can't create an instance method since these are only used as defaults to another instance method)

Reply:
I like it